addons/mail/static/src/core/common/plugin/mention_plugin.js

```diff
@@ -24,6 +24,14 @@ export class MentionPlugin extends Plugin {
                     this.dependencies.history.addStep();
                 },
             },
+            {
+                selector: "a.o_message_redirect",
+                checker: (el) => this.isValidMessageMentionElement(el),
+                validMentionsHandler: (messageLinks) => {
+                    this.store.handleValidMessageMention(messageLinks);
+                    this.dependencies.history.addStep();
```

Comment:
This might add an history step even if `handleValidMessageMention` does nothing.
We might want to return whether it did something or not.